lib/regressions/multivariableLinearRegression.ts

```diff
@@ -1,55 +1,33 @@
 import { meanSquaredError } from '../functions/losses'
 import { gradientDescent } from '../functions/optimizers'
-
-export type MultivariableLinearRegressionOutputType = {
-    updatedWeight: number[]
-    updatedBias: number
-    costHistory: number[]
-}
-
-export function predictionMultivariable(input: number[][], weight: number[], bias: number): number[] {
-    const value: number[] = []
-
-    input.forEach((feature) => {
-        let pred = 0
-        for (let i = 0; i < input[0].length; i++) {
-            pred += feature[i] * weight[i]
-        }
-        value.push(pred + bias)
-    })
-
-    return value
-}
+import { predictMultivariable, RegressionOutputType, transposeAndNormalize } from './utilities'
 
 export default class MultivariableLinearRegression {
     static *fit(
         input: number[][],
         target: number[],
-        weight: number[],
-        bias: number,
         learningRate: number,
         epochs: number,
         costFunction: Function
-    ): Generator<MultivariableLinearRegressionOutputType> {
+    ): Generator<RegressionOutputType> {
         const costHistory = []
-        let updatedWeight = weight
-        let updatedBias = bias
-        let updatedPrediction = predictionMultivariable(input, weight, bias)
+        const transformedInput = transposeAndNormalize(input)
+        let biasAndWeights = Array(input[0].length + 1).fill(0)
         let currentEpoch = 0
+        let updatedPrediction = predictMultivariable(transformedInput, biasAndWeights)
 
         while (true) {
```

Comment:
is there a stopping condition?

Reply:
Should there be a stopping condition? There I initialize the variable `updatedPrediction`.